fission/function.go

```diff
@@ -497,3 +536,43 @@ func fnPods(c *cli.Context) error {
 
 	return err
 }
+
+func fnTest(c *cli.Context) error {
+	fnName := c.String("name")
+	routerURL := os.Getenv("FISSION_ROUTER")
+	if len(routerURL) == 0 {
+		fatal("Need FISSION_ROUTER set to your fission router.")
+	}
+
+	url := fmt.Sprintf("http://%s/fission-function/%s", routerURL, fnName)
+
+	maxRetries := 3
+	retryDelay := 100 * time.Millisecond
+	var resp *http.Response
+	fnReachable := false
+
+	for i := 0; i < maxRetries; i++ {
+		resp = httpRequest(c.String("method"), url, c.String("body"), c.StringSlice("header"))
+		if resp.StatusCode < 300 {
+			body, err := ioutil.ReadAll(resp.Body)
+			checkErr(err, "Function test")
+			fmt.Print(string(body))
+			defer resp.Body.Close()
+			fnReachable = true
+			break
+		}
+		time.Sleep(retryDelay)
+	}
+
+	if !fnReachable {
+		body, err := ioutil.ReadAll(resp.Body)
+		checkErr(err, "Function test failed")
```

Comment:
See other uses of checkErr -- this doesn't fit the grammar of the error message. Also, this error is not about the function test failing, it's about failing to read the http response